src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,132 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int MAX_LENGTH = 1073741824;
+    private static final int MULTIPLY_SIZE = 2;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putKeyValue(null, value);
+            return;
+        }
+
+        int hash = hash(key);
+        int index = hash & (table.length - 1);
+        Node<K, V> current = table[index];
 
+        while (current != null) {
+            if (key.equals(current.key)) {
+                current.value = value;
+                return;
+            }
+            current = current.next;
+        }
+        table[index] = new Node<>(hash, key, value, table[index]);
+        size++;
+        resizeIfNeeded();
+    }
+
+    private void putKeyValue(K key, V value) {
```

Comment:
I believe you could avoid this method,in method 'put' if key is null assume that index is 0, it will reduce code duplication